### PR TITLE
refactor: allow builders to accept emoji strings

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Review by kyranet — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

File: packages/discord.js/typings/index.test-d.ts [hunk 1]
```diff
@@ -1358,7 +1358,11 @@ new SelectMenuComponent({
 
 new ButtonComponent({
   style: ButtonStyle.Danger,
-});
+})
+  .setEmoji('<a:foo:123>')
+  .setEmoji('<:foo:123>')
+  // @ts-expect-error
+  .setEmoji(':foobar:123');
```
Comment (vladfrangu) [inline packages/discord.js/typings/index.test-d.ts:1365]: Util.parseEmoji can also match `foobar:123`, `:foobar:123`, etc. Might want to update types if we want to be extra strict, or just keep it as string

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/discord.js/typings/index.test-d.ts (hunk 1 above)
Comment (imranbarbhuiya) [reply packages/discord.js/typings/index.test-d.ts:1365]: It should allow default emojis too. So imo string will be better option.

Comment (ImRodry) [reply packages/discord.js/typings/index.test-d.ts:1365]: It should just allow everything that parseEmoji allows, just like in v13

Comment (imranbarbhuiya) [reply packages/discord.js/typings/index.test-d.ts:1365]: > It should just allow everything that parseEmoji allows, just like in v13
It should also allow `APIMessageComponentEmoji`.

File: packages/discord.js/src/structures/SelectMenuOption.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+const { SelectMenuOption: BuildersSelectMenuOption } = require('@discordjs/builders');
+const Util = require('../util/Util');
+
+/**
+ * Represents a select menu option
+ */
+class SelectMenuOption extends BuildersSelectMenuOption {
+  /**
+   * Sets the emoji to display on this option
+   * @param {APIMessageComponentEmoji} emoji The emoji to display on this option
```
Comment (muchnameless) [inline packages/discord.js/src/structures/SelectMenuOption.js:12]:
```suggestion
   * @param {ComponentEmojiResolvable} emoji The emoji to display on this option
```

**Review by muchnameless — CHANGES_REQUESTED** [review]

File: packages/discord.js/typings/index.test-d.ts [hunk 3]
```diff
@@ -1367,7 +1367,16 @@ new SelectMenuBuilder({
 
 new ButtonBuilder({
   style: ButtonStyle.Danger,
-});
+})
+  .setEmoji('<a:foo:123>')
+  .setEmoji('<:foo:123>')
+  .setEmoji('foobar:123')
+  .setEmoji('normal')
```
Comment (vladfrangu) [inline packages/discord.js/typings/index.test-d.ts:1374]:
```suggestion
  .setEmoji('😏')
```

**Review by vladfrangu — APPROVED** [review]
LGTM, just add this

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**